src/common/meta/src/ddl/create_table.rs

```diff
@@ -265,16 +265,38 @@ impl CreateTableProcedure {
     /// - Failed to create table metadata.
     async fn on_create_metadata(&mut self) -> Result<Status> {
         let table_id = self.table_id();
+        let cluster_id = self.creator.data.cluster_id;
         let manager = &self.context.table_metadata_manager;
 
         let raw_table_info = self.table_info().clone();
         // Safety: the region_wal_options must be allocated.
         let region_wal_options = self.region_wal_options()?.clone();
         // Safety: the table_route must be allocated.
-        let table_route = TableRouteValue::Physical(self.table_route()?.clone());
+        let physical_table_route = self.table_route()?.clone();
+
+        let idents = physical_table_route
+            .region_routes
+            .iter()
+            .map(|region| {
+                (
+                    cluster_id,
+                    region.leader_peer.as_ref().unwrap().id,
+                    region.region.id,
+                )
+            })
+            .collect::<Vec<_>>();
+        let table_route = TableRouteValue::Physical(physical_table_route);
         manager
             .create_table_metadata(raw_table_info, table_route, region_wal_options)
             .await?;
+        // Notifies RegionSupervisor to register failure detector of new created regions.
+        //
+        // The datanode may crash without sending a heartbeat that contains information about newly created regions,
+        // which may prevent the RegionSupervisor from detecting failures in these newly created regions.
+        self.context
```

Comment:
I'd like to wrap these logic into a function like `register_region_failure_detectors`, and make the `on_create_metadata` clean and smaller.